torch_xla/csrc/aten_autograd_ops.cpp

```diff
@@ -141,6 +151,15 @@ torch::autograd::variable_list MaxPool2dAutogradFunction::backward(
       c10::DispatchKey::Functionalize,
   });
   auto ks = self_keyset & mask;
+  // If python dispatcher is enabled, we need to hit it.
+  // This is a bit hacky, we should probably come up with
+  // a better way to do this (maybe don't redispatch?)
+  if (c10::impl::PythonDispatcherTLS::get_state()) {
+    ks = ks.add(c10::DispatchKey::PythonDispatcher);
+  }
+  if (ks.has(c10::DispatchKey::Python)) {
+    ks = ks.add(c10::DispatchKey::PythonTLSSnapshot);
+  }
```

Comment:
do we need to do this for all functions in this file? including `MaxPool3dAutogradFunction ` and `einsum`

Reply:
ah yep, I think we will. It looks like einsum and maxpool3d don't have any of the existing logic either (computing a dispatch keyset and manually reinvoking the dispatcher), so I'm guessing it's just the case that nobody has tried to used those ops with torch.compile yet.

Reply:
lol I just added a test for einsum today with `torch.compile`, should I be worried?

Reply:
Yeah…. I would naively expect it to break unless I’m missing something. I could be wrong, lmk if you hit something :)

Reply:
what's the right way for me to test it? build your branch locally?